pennylane/fermi/fermionic.py

```diff
@@ -154,23 +152,19 @@ def __add__(self, other):
         if isinstance(other, FermiWord):
             return self_fs + FermiSentence({other: 1.0})
 
-        if isinstance(other, (Number, ndarray)):
-            if isinstance(other, ndarray) and qml.math.size(other) > 1:
-                raise ValueError(
-                    f"Arithmetic Fermi operations can only accept an array of length 1, "
-                    f"but received {other} of length {len(other)}"
-                )
-            return self_fs + FermiSentence({FermiWord({}): other})
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot add {type(other)} to a FermiWord.")
+        if qml.math.size(other) > 1:
```

Comment:
Just add a newline between the `isisntance` check and the `size` assertion here and elsewhere.